apps/ledger-live-mobile/src/screens/Swap/Form/Connect.tsx

```diff
@@ -33,6 +36,28 @@ export function Connect({
     }
   }, [result, setResult]);
 
+  const navigation = useNavigation();
+
+  // Only reacts on an update request for the left part of the header
+  // Keeping the rest of the header (exchange and history tab + close button) from the exchange screen.
+  const requestToSetHeaderOptions = useCallback(
+    (request: SetHeaderOptionsRequest) => {
+      if (request.type === "set") {
+        // SwapForm, rendering Connect, is wrapped in a Tab.Screen. To update the left header,
+        // we need to set the options of the parent.
+        navigation.getParent()?.setOptions({
+          headerLeft: request.options.headerLeft,
+        });
+      } else {
+        // Sets back the left part of the header to its initial values
+        navigation.getParent()?.setOptions({
+          headerLeft: () => null,
+        });
+      }
+    },
+    [navigation],
+  );
+
```

Comment:
maybe here you need to ensure that on unmount you cleanup the options of the navigation parent?

Reply:
You're right, going to add some cleaning on unmount to be safe ;)